src/couch_js/src/couch_js.erl

```diff
@@ -49,3 +52,15 @@ map_docs(Proc, Docs) ->
         end, Results),
         {Doc#doc.id, Tupled}
     end, Docs)}.
+
+acquire_context() ->
+    Ctx = couch_query_servers:get_os_process(?JS),
+    {ok, Ctx}.
+
+
+release_context(Proc) ->
+    couch_js_query_servers:ret_os_process(Proc).
+
+
+try_compile(Proc, FunName, FunSrc) ->
+    couch_query_servers:try_compile(Proc, map, FunName, FunSrc).
```

Comment:
Is it always would be a `map` in a second argument?

Reply:
I can see that we used to call `couch_query_servers:try_compile(Proc, reduce, RedName, RedSrc)` [here](https://github.com/apache/couchdb/pull/3481/files#diff-637f0ee4524b44455ad1fcbfa99fdbdc757c0a0c7a1bd462f4e9b5f51a548dc9L206)

Reply:
Looks like we either need to pass it explicitly as try_compile argument or have it as part of a context.

Reply:
Never mind the idea of passing the `map | reduce` in the context. I can see now that we are using both in `couch_mrview:validate/3`

- map on [line 192](https://github.com/apache/couchdb/pull/3481/files#diff-637f0ee4524b44455ad1fcbfa99fdbdc757c0a0c7a1bd462f4e9b5f51a548dc9L192)
- reduce on [line 206](https://github.com/apache/couchdb/pull/3481/files#diff-637f0ee4524b44455ad1fcbfa99fdbdc757c0a0c7a1bd462f4e9b5f51a548dc9L206)